src/libs/CMakeLists.txt

```diff
@@ -92,7 +92,8 @@ if(PortMidi_FOUND)
   target_link_libraries (midi ${PortMidi_LIBRARY})
 endif()
 
-find_package(SDL2)
+if (USE_SDL2)
+find_package(SDL2 REQUIRED)
```

Comment:
This makes SDL2 no longer truly optional (auto-enabled), and will make configure fail. To be discussed if the changed behaviour is what is wanted.

Reply:
I'd just remove REQUIRED to keep current behaviour, but still have the new option to disable SDL2

Reply:
Having it always optional makes it hard to notice when things break (SDL not detected but wanted). But this will require 3-way setting on/off/auto, if we want to keep auto-detection feature (which I’m not really sure to be useful).

Reply:
Autodetect is a dangerous option IMO, as it does not reflect the clear intent of the person compiling the program.

Reply:
Sure, providing feedback to the user this way can be useful.

Just saying that none of the "optional" options we have today work this way, so if this is something we want, we need to change how they're all managed.

Reply:
I've added the `AUTODETECT_SDL2` to keep current behavior, which is on by default. With `USE_SDL2` distributions can use explicit behavior.
Does this look reasonable?

Reply:
Sorry, but I'm afraid it's overcomplicating it IMHO.

I'd rather you either keep the USE_SDL2 and remove REQUIRED to match the current options behaviour, or make a proposal to refactor all of them together, so there are no exceptions or surprises.

Reply:
@kmilos sure, I've updated the changes again to add/use  `USE_SDL2` only without `REQUIRED`. I guess this is fine!